vllm/model_executor/layers/quantization/kernels/mixed_precision/xla.py

```diff
@@ -0,0 +1,85 @@
+from typing import Optional, Tuple
+
+import torch
+
+from vllm.scalar_type import scalar_types
+
+from .MPLinearKernel import MPLinearKernel, MPLinearLayerConfig
+
+XLA_SUPPORTED_QUANT_TYPES = [scalar_types.uint8b128]
```

Comment:
qq: what's the difference between `scalar_types.int8` and `scalar_types.uint8b128`?

Reply:
`int8` represents negative numbers using 2s complement (just like C/C++ `int8`) but `uint8b128` is an unsigned 8bit integer with a bias of 128, i.e. its a number in the range 0-255 but during inference 128 should be subtracted off so it represents the range -128 to 127. This is a common pattern in 4bit inference where `uint4b8` (an unsigned 4bit integer with a bias of 8) is generally preferred over `int4` since its more efficient to upconvert.

Reply:
Thanks @LucasWilkinson for the explanation.

So this means the weight tensor will be in `scalar_types.uint8b128`? Then I think post-processing logic in `transform_w_int8(x)` would not be correct?

Reply:
The uint8b128 is basically int8 packed into int32. So the transform is trying to convert this dtype into int8 (as required by xla) 

the models are serialized in this way due to annoying historical  reasons 😢

this is not working quite right for w8a16

Reply:
@robertgshaw2-neuralmagic do you know if compressed tensor stores as `uint8b128` or `int8`? It appears from the existing code `uint8b128`, if that is the case to go from packed `uint8b128` we'd have to do something like:
```
        def transform_w_int8(x):
             # reinterpet_cast the packed int32 weights as int8
             # convert to [int, out] -> [out, int]
             return (x.view(dtype=torch.uint8).to(torch.int16) - 128).to(torch.int8).t()
```

basically if I had the values:
`[  0,   1,  -1,  -2,   2,  17, -28,  72]`
for `uint8b128` these would be encoded as:
```
tensor([128, 129, 127, 126, 130, 145, 100, 200], dtype=torch.uint8)
```
for `int8` 
```
tensor([  0,   1,  -1,  -2,   2,  17, -28,  72], dtype=torch.int8)
```

(so to convert `uint8b128`, `int32` packed, to `int8` you wold do something like:`(x.view(dtype=torch.uint8).to(torch.int16) - 128).to(torch.int8)`, i.e. reinterpret to `torch.uint8` then upcast to `int16` so we can subtract the bias, subtract the 128 bias then cast to `int8`)

e.g.
```
>>> x
tensor([2122285440, -932933246], dtype=torch.int32)
>>> x.view(torch.uint8)
tensor([128, 129, 127, 126, 130, 145, 100, 200], dtype=torch.uint8)
>>> (x.view(dtype=torch.uint8).to(torch.int16) - 128).to(torch.int8)
tensor([  0,   1,  -1,  -2,   2,  17, -28,  72], dtype=torch.int8)
```


(not sure why it would go OOM though)

(cc @dsikka @mgoin , do you know if compressed tensor is stored unsigned with a bias?)